### PR TITLE
Hide column

### PR DESCRIPTION
**Pull request by rmourad10**
Pull request to hide columns during extensive log display. The feature should help to block a certain number of columns if a user specifies log data.

### CODE REVIEW

**Comment by alankilborn** [comment]
PRs should be linked to an issue where the proposed functionality is discussed and possibly approved by the author of Notepad++.

This PR could (and should) be implemented as a series of macros, by individual users that need this functionality.

**Comment by alankilborn** [comment]
Example macro:

        <Macro name="Delete Columns 1-10" Ctrl="no" Alt="no" Shift="no" Key="0">
            <Action type="3" message="1700" wParam="0" lParam="0" sParam="" />
            <Action type="3" message="1601" wParam="0" lParam="0" sParam="(?-s)^.{1,10}(.*)" />
            <Action type="3" message="1625" wParam="0" lParam="2" sParam="" />
            <Action type="3" message="1602" wParam="0" lParam="0" sParam="${1}" />
            <Action type="3" message="1702" wParam="0" lParam="768" sParam="" />
            <Action type="3" message="1701" wParam="0" lParam="1609" sParam="" />
        </Macro>

File: PowerEditor/src/Notepad_plus.cpp [hunk 1]
```diff
@@ -725,12 +725,73 @@ LRESULT Notepad_plus::init(HWND hwnd)
...
 
+	void hideColumns(int numColumns) {
+    HWND editorHandle = ::GetDlgItem(_hSelf, IDC_MAIN_EDIT);
+    int lineCount = ::SendMessage(editorHandle, SCI_GETLINECOUNT, 0, 0);
+
+    for (int line = 0; line < lineCount; ++line) {
+        int lineStart = ::SendMessage(editorHandle, SCI_POSITIONFROMLINE, line, 0);
+        ::SendMessage(editorHandle, SCI_SETTARGETSTART, lineStart, 0);
+        ::SendMessage(editorHandle, SCI_SETTARGETEND, lineStart + numColumns, 0);
+        ::SendMessage(editorHandle, SCI_REPLACETARGET, 0, reinterpret_cast<LPARAM>(""));
```
Comment (donho) [inline PowerEditor/src/Notepad_plus.cpp:742]: It's rather "Remove" the column than "Hide" the column.

Comment (alankilborn) [reply PowerEditor/src/Notepad_plus.cpp:742]: and that's why I suggested a macro with the name `Delete Columns...`

**Comment by Yaron10** [comment]
@rmourad10,

> Have you even bothered to compile your PR ?

This offensive and condescending question is outrageous coming from someone who usually does not bother to test his own work.

Here are some recent examples:

https://github.com/notepad-plus-plus/notepad-plus-plus/commit/a51e62c16dc45160afe6c7deb19096e98ad18372.
Does not fix anything.

https://github.com/notepad-plus-plus/notepad-plus-plus/commit/950236bb8f8e85d678912939c00183e85c6417f6.
Was that even tested?

https://github.com/notepad-plus-plus/notepad-plus-plus/commit/2724e0ddbb1e7349fd9c23711d6b4e9535bcf1ba.
Major bugs and regressions.

**Comment by donho** [comment]
@Yaron10 
I'm sorry to not be able to apply your vision for the [RTL implementation](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14431#issuecomment-1836301467). Though I do believe your use case is not the common case of the RTL usage, it's been said that I'm opened for reconsidering your suggestion in the future, if the similar requests is coming.
I suggest you create an issue (a feature request) so we can monitor it.

I always test my code, otherwise it's impossible that Notepad++ is used by people. It could be, I don't test as thoroughly as you do - that's why I appreciate to work with you: it’s impressive that you always find the unusual producible bugs and corner cases insightfully. 

> Here are some recent examples:

>https://github.com/notepad-plus-plus/notepad-plus-plus/commit/a51e62c16dc45160afe6c7deb19096e98ad18372.
Does not fix anything.

>https://github.com/notepad-plus-plus/notepad-plus-plus/commit/950236bb8f8e85d678912939c00183e85c6417f6.
Was that even tested?

>https://github.com/notepad-plus-plus/notepad-plus-plus/commit/2724e0ddbb1e7349fd9c23711d6b4e9535bcf1ba.
Major bugs and regressions.

As I said, and I say it again, all the bug reports are welcome, and I engage to fix them. 
But if you do as some bug reports do: *There is a bug, I cannot use it.*, then I can do nothing about it.
